src/OrchardCore.Modules/OrchardCore.AdminDashboard/Services/DashboardPartContentTypeDefinitionHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System.Text.Json.Nodes;
+using OrchardCore.AdminDashboard.Models;
+using OrchardCore.ContentManagement.Metadata.Records;
+using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.ContentTypes.Events;
+using OrchardCore.Modules;
+
+namespace OrchardCore.AdminDashboard.Services;
+
+public sealed class DashboardPartContentTypeDefinitionHandler : ContentDefinitionHandlerBase
+{
+    public override void BuildingContentType(BuildingContentTypeContext context)
```

Comment:
Why not do this in `ContentTypeUpdated` and `ContentTypeCreated`?

Reply:
The `ContentTypeUpdated` is called when `ContentDefinitionService.AlterTypePartsOrderAsync` method is called.
ContentTypeCreated is called when ContentDefinitionService.AddTypeAsync method is called. These methods are only called by the ContentTypes controller. 

We need to hook into the building event which happens when the content definition is build just before it is cached. The new events allow us to hook into that level.

Reply:
OK then!